File: CPlusPlus/ExampleTests/test_calculator.cpp
#include <gtest/gtest.h>
#include <iostream>
#include "common/common.h"
#include "calculator_glue.h"

TEST(Calculator, Calculation_AddTwoNumbers) {
    CalculatorGlue glue;
    std::vector<AdderString> objectList1 = {
        AdderString::from_vec({"2", "3", "5"}),
        AdderString::from_vec({"10", "20", "30"}),
        AdderString::from_vec({"-1", "1", "0"}),
    };
    glue.examples_calculation_add_two_numbers(objectList1);
}

File: Go/ExampleTests/test_calculator.go
package calculator

import (
	"testing"
	"calculator/common"
)

func TestCalculation_AddTwoNumbers(t *testing.T) {
	glue := NewCalculatorGlue()
	objectList1 := []common.AdderString{
		common.NewAdderStringFromSlice([]string{"2", "3", "5"}),
		common.NewAdderStringFromSlice([]string{"10", "20", "30"}),
		common.NewAdderStringFromSlice([]string{"-1", "1", "0"}),
	}
	glue.ExamplesCalculationAddTwoNumbers(t, objectList1)
}

